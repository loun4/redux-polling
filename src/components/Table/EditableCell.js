import React from 'react';
import propTypes from 'prop-types';

const EditableCell = ({ value, onChange, onFocus }) => {
  const handleChange = e => onChange(Number(e.target.value));

  return (
    <td>
      <input
        type="number"
        step=".01"
        value={value}
        onChange={handleChange}
        onFocus={onFocus}
      />
    </td>
  );
};

EditableCell.propTypes = {
  value: propTypes.number.isRequired,
  onChange: propTypes.func.isRequired,
  onFocus: propTypes.func,
};

EditableCell.defaultProps = {
  onFocus: null,
};

export default EditableCell;
