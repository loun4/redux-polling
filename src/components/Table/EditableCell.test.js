import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import EditableCell from './EditableCell';

const noop = () => {};

const buildSubject = (onFocus = noop, onChange = noop) => {
  return <EditableCell value={1} onFocus={onFocus} onChange={onChange} />;
};

test('Render EditableCell', () => {
  const tree = renderer.create(buildSubject()).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Call onChange', () => {
  const mockHandler = jest.fn();
  const wrapper = shallow(buildSubject(undefined, mockHandler));

  wrapper
    .find('input')
    .props()
    .onChange({
      target: {
        value: '5',
      },
    });

  expect(mockHandler).toHaveBeenCalledWith(5);
});

test('Call onFocus', () => {
  const mockHandler = jest.fn();
  const wrapper = shallow(buildSubject(mockHandler));

  wrapper
    .find('input')
    .props()
    .onFocus();

  expect(mockHandler).toHaveBeenCalled();
});
