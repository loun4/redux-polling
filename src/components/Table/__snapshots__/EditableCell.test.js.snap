// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EditableCell 1`] = `
<td>
  <input
    onChange={[Function]}
    onFocus={[Function]}
    step=".01"
    type="number"
    value={1}
  />
</td>
`;
